tests/test_association_db_sqlite.py

```diff
@@ -151,251 +156,149 @@ def _compare_source_records(self, record_a, record_b):
-        indexer_ids = np.array(
-            self.assoc_db._db_cursor.fetchall(), np.int).flatten()
+        dia_objects = create_test_points(
+            n_points=n_objects,
+            start_id=0,
+            schema=make_minimal_dia_object_schema(),
+            point_locs_deg=[[0.0, 0.0] for idx in range(n_objects)],
+            scatter_arcsec=1.0,
+            indexer_ids=[0 for idx in range(n_objects)],)
 
-        dia_object_catalog = self.assoc_db._get_dia_object_records(indexer_ids)
+        self.assoc_db.store_dia_objects(dia_objects, False)
 
-        for obj_idx, dia_object_record in enumerate(dia_object_catalog):
-            self._compare_source_records(
-                dia_object_record,
-                dia_collection.dia_objects[obj_idx].dia_object_record)
+        output_dia_objects = self.assoc_db._get_dia_object_catalog([0])
+        for dia_object, created_object in zip(output_dia_objects, dia_objects):
+            self._compare_source_records(dia_object, created_object)
 
-    def test_get_dia_sources(self):
-        """ Test the retrieval of DIASources from the database.
+    def test_store_and_load_dia_sources(self):
+        """Test the retrieval of DIASources from the database.
         """
-        dia_objects = create_test_dia_objects(
-            n_objects=1, n_sources=5)
-        dia_collection = DIAObjectCollection(dia_objects)
-
-        self.assoc_db.store(dia_collection, True)
+        n_sources = 5
+        dia_sources = create_test_points(
+            n_points=n_sources,
+            start_id=0,
+            schema=make_minimal_dia_source_schema(),
+            point_locs_deg=[[0.1, 0.1] for idx in range(n_sources)],
+            scatter_arcsec=1.0,
+            associated_ids=[0 for idx in range(n_sources)])
 
-        self.assoc_db._commit()
+        self.assoc_db.store_dia_sources(dia_sources, [0 for idx in range(n_sources)])
 
-        src_cat = self.assoc_db._get_dia_sources(0)
-        for dia_source, created_source in zip(
-                src_cat, dia_collection.dia_objects[0].dia_source_catalog):
+        src_cat = self.assoc_db.load_dia_sources([0])
+        for dia_source, created_source in zip(src_cat, dia_sources):
             self._compare_source_records(dia_source, created_source)
 
-    def test_store_dia_object_dia_source_pair(self):
-        """ Test storing the ids of associated DIAObjects and DIASources.
-        """
-        for obj_id in range(2):
-            for src_id in range(5):
-                self.assoc_db._store_dia_object_source_pair(
-                    obj_id, src_id + (obj_id * 5))
-        self.assoc_db._commit()
+        dia_sources = create_test_points(
+            n_points=n_sources,
+            start_id=5,
+            schema=make_minimal_dia_source_schema(),
+            point_locs_deg=[[0.0, 0.0] for idx in range(n_sources)],
+            scatter_arcsec=1.0,
+            associated_ids=[1 for idx in range(n_sources)])
```

Comment:
Again, try not to have test "cases" that are really chains of independent tests.